### PR TITLE
Fix issue

### LINKED ISSUE #201: Error showing Mutated genes and CNA genes tables
CNA genes table is not shown for 'brca_metabric' study

### PR DESCRIPTION
**Pull request by kalletlak**
Fix #201, #196, #203

### CODE REVIEW

File: app/scripts/model/dataProxy.js [hunk 1]
```diff
@@ -703,9 +706,9 @@ window.DataManagerForIviz = (function($, _) {
             });
           $.when.apply($, requests).then(function() {
             _.each(_profiles, function(_profile) {
-              if (_profile.study_id + '_gistic' === _profile.id) {
-                self.gisticProfileIdsMap[_profile.study_id] = _profile.id;
-              } else if (_profile.study_id + '_mutations' === _profile.id) {
+              if (_profile.genetic_alteration_type === 'COPY_NUMBER_ALTERATION') {
```
Comment (zhx828) [inline app/scripts/model/dataProxy.js]: There are two COPY_NUMBER_ALTERATION for hnsc_tcga

File: app/scripts/views/components/dataTable/tableView.js [hunk 2]
```diff
@@ -164,26 +164,25 @@
       selectedGeneData.length = 0;
 
       if (geneData_) {
-        _.each(geneData_, function(item) {
+        _.each(geneData_, function(item, index) {
```
Comment (zhx828) [inline app/scripts/views/components/dataTable/tableView.js:167]: Is there any issues wich previous implementation?

Comment (kalletlak) [reply app/scripts/views/components/dataTable/tableView.js:167]: I need index for each gene data, which is going to be unique id for the table row data.

File: app/scripts/model/dataProxy.js (hunk 1 above)
Comment (kalletlak) [reply app/scripts/model/dataProxy.js]: Once the issue is fixed(https://github.com/cBioPortal/cbioportal/issues/1645), I'll update this.

It should be like 
if(_profile.genetic_alteration_type === 'COPY_NUMBER_ALTERATION' && _profile.show_profile_in_analysis_tab){....}

**Pull request merged**